cl/alerts/tasks.py

```diff
@@ -28,117 +30,142 @@ def enqueue_docket_alert(d_pk: int) -> bool:
     return create_redis_semaphore("ALERTS", key, ttl=60 * 10)
 
 
+def get_docket_alert_recipients(
+    d_pk: int,
+    recap_email_recipients: list[str] = [],
+) -> tuple[
+    list[tuple[str, str]],
+    list[tuple[str, str]],
+    list[tuple[str, str]],
+    list[str],
```

Comment:
OK, this refactor isn't quite it, and this return type should have been the "[code-smell](https://en.wikipedia.org/wiki/Code_smell)" that made you realize something was wrong. 

Linus Torvalds once said something like, "when you've got your data objects sorted out and as simple as possible, you know you're done." This ain't that. 

The solution here is hinted at by the name of the function. We want to get recipients, so it's going to be a list. But a list of what? We could use dicts like:

```
email: 'mike@free.law',
secret_key: 'asdfasdfasdfa',
first_time: True,
auto_subscribe: False,
does_not_exist: False,
```

Or, dare I say it: We could use an object to hold that information (this would make typing easier).

Then the return from this function would be something like `list(MessageData)` (or whatever). This is all notional, meaning I want you to make it good, but this return type is a major code smell.

Reply:
Oh, I guess a data class is probably a good tool here too: https://docs.python.org/3/library/dataclasses.html

Reply:
I imagined I should have to use something like this, but I wasn't sure what kind of data structure was appropriate so I ended up using that ugly list of tuples. The new data class `DocketAlertRecipient` helps simplify all the things here and is clearer.

I just keep the `recap_email_user_does_not_exist_list` because is just a list of emails so it doesn't match with the new data class.

Thank you!

Reply:
Looks good. Thanks again.